File: oop/Warning.java

public class Warning {
  public static void main(String args[]) {
    String s = (String) "hello, word";
    System.out.println(s);
  }
}

// Using -Xlint:
// $ javac -Xlint Warning.java
// Warning.java:4: warning: [cast] redundant cast to String
//     String s = (String)"hello, word!";
//                ^
// 1 warning

// Using -Xlint and -Werror:
// $ javac -Xlint -Werror Warning.java
// Warning.java:4: warning: [cast] redundant cast to String
//     String s = (String)"hello, word";
//                ^
// error: warnings found and -Werror specified
// 1 error
// 1 warning

File: oop/warning.cc

#include <iostream>

using namespace std;

int main(int argc, char* argv[])
{
  int i;
  cout << i << endl;
}

// Using -Wall:
// $ g++ -Wall warnings.cpp
// warnings.cpp: In function ‘int main(int, char**)’:
// warnings.cpp:10:11: warning: ‘i’ is used uninitialized in this function [-Wuninitialized]
//    cout << i << endl;
//            ^

// Using -Wall and -Werror:
// $ g++ -Wall -Werror warnings.cpp
// warnings.cpp: In function ‘int main(int, char**)’:
// warnings.cpp:10:11: error: ‘i’ is used uninitialized in this function [-Werror=uninitialized]
//    cout << i << endl;
//            ^
// cc1plus: all warnings being treated as errors
